src/driver/hermes_mod.h

```diff
@@ -79,6 +89,47 @@ struct __attribute__((__packed__)) hermes_cfg {
 	uint32_t ehdssze;
 };
 
+struct __attribute__((__packed__)) hermes_cmd_req {
+	uint8_t opcode;
+	uint8_t rsv0;
+	uint16_t cid;
+	uint32_t rsv1;
+	union {
+		struct __attribute__((__packed__)) {
+			uint8_t prog_slot;
+			uint8_t data_slot;
+		} run_prog;
+
+		uint8_t cmd_specific[24];
+	};
+};
+
+struct __attribute__((__packed__)) hermes_cmd_res {
+	uint16_t cid;
+	uint8_t status;
+	uint8_t rsv0[5];
+	union {
+		struct __attribute__((__packed__)) {
+			uint64_t ebpf_ret;
+		} run_prog;
```

Comment:
Do we really need this abstraction at this point? Do we ever have plans to create non-ebpf program runners? Are you sure such things will not require the same arguments? I wouldn't predict the future and skip the union. Adding some reserved space is a good idea, but no need to do much else.

Reply:
Fair point. I'll remove it from the next iteration.